flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/source/hybrid/HybridSourceSplitEnumerator.java

```diff
@@ -304,6 +304,7 @@ public Object getPreviousEnumerator() {
         LOG.info("Starting enumerator for sourceIndex={}", currentSourceIndex);
         context.setIsProcessingBacklog(currentSourceIndex < sources.size() - 1);
         currentEnumerator.start();
+        finishedReaders.clear();
```

Comment:
`finishedReaders` is used to track readers that have finished processing for current enumerator, but the `  currentEnumerator` here has been switched to next enumerator and the next enumerator has been started which may lead to some potential issue, I think we should add the cleanup piece following line 251, the resource cleanup should follow its serving enumerator, right?

Reply:
I guess you mean line 261?

https://github.com/apache/flink/blob/c1be70bbc60e4271d9ff80ac4eaf3421d7602058/flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/source/hybrid/HybridSourceSplitEnumerator.java#L261

Line 251 is for closing the HybridEnumerator itself, so whether to cleanup the finishedReaders is not critial, as we will not use this hybrid enumerator any more.

Reply:
Yes, @Apache9 line 261, my typo.